editor/plugins/collision_shape_2d_editor_plugin.cpp

```diff
@@ -176,16 +177,26 @@ void CollisionShape2DEditor::set_handle(int idx, Point2 &p_point) {
 		} break;
 
 		case RECTANGLE_SHAPE: {
-			if (idx < 3) {
+			if (idx < 8) {
 				Ref<RectangleShape2D> rect = node->get_shape();
+				Vector2 size = (Point2)original;
 
-				Vector2 size = rect->get_size();
-				if (idx == 2) {
-					size = p_point * 2;
+				if (RECT_HANDLES[idx].x != 0) {
+					size.x = p_point.x * RECT_HANDLES[idx].x * 2;
+				}
+				if (RECT_HANDLES[idx].y != 0) {
+					size.y = p_point.y * RECT_HANDLES[idx].y * 2;
+				}
+
+				if (!Input::get_singleton()->is_key_pressed(KEY_ALT)) {
```

Comment:
Minor nitpick, but shouldn't we put the affirmative condition first? `if (Input::get_singleton()->is_key_pressed(KEY_ALT))`